pkg/scheduler/framework/plugins/interpodaffinity/scoring.go

```diff
@@ -203,14 +203,12 @@ func (pl *InterPodAffinity) PreScore(
 	var antiAffinityTerms []*weightedAffinityTerm
 	if hasAffinityConstraints {
 		if affinityTerms, err = getWeightedAffinityTerms(pod, affinity.PodAffinity.PreferredDuringSchedulingIgnoredDuringExecution); err != nil {
-			klog.Error(err)
-			return nil
+			framework.NewStatus(framework.Error, fmt.Sprintf("processing affinity terms: %+v", err))
```

Comment:
same question as the other PR. However, I'm not sure if the framework can handle UnschedulableAndUnresolvable properly in 1.18 (I think it just converts it to Error anyways). Could you check?

Reply:
Good point, checked it out and responded in https://github.com/kubernetes/kubernetes/pull/93660#discussion_r466610419

>`UnschedulableAndUnresolvable` seems to only be parsed in Filter plugins, and the [error code definitions](https://github.com/kubernetes/kubernetes/blob/release-1.19/pkg/scheduler/framework/v1alpha1/interface.go#L59) imply that unexpected input could cause an error.
>
>So `UnschedulableAndUnresolvable` could work for the Filter plugin, but score just returns anything that's not success as an error (https://github.com/kubernetes/kubernetes/blob/release-1.19/pkg/scheduler/core/generic_scheduler.go#L512-L516) (also, in 1.18 [this case is already returned as a `framework.Error`](https://github.com/kubernetes/kubernetes/blob/release-1.18/pkg/scheduler/framework/plugins/interpodaffinity/filtering.go#L350) from the Filter plugin)
>
>I think in the case of a syntax error, `Error` makes sense. `UnschedulableAndUnresolvable` sounds more related to cluster state that prevents the pod from scheduling (ie, it fails the filters and nothing can be preempted to fit it)